core/src/main/scala/chisel3/reflect/DataMirror.scala

```diff
@@ -55,7 +55,7 @@ object DataMirror {
     * @param x the `Data` to check
     * @return `true` if x is a Probe, `false` otherwise
     */
-  def hasProbeTypeModifier(x: Data): Boolean = x.probeInfo.nonEmpty
+  private[chisel3] def hasProbeTypeModifier(x: Data): Boolean = x.probeInfo.nonEmpty
```

Comment:
this is main user facing API change, this needs to be replaced with explicit matches on the ProbeLike and friends